web-library/Book/Entity/BookCopy.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace web_library.Book.Entity;
+
+public class BookCopy
+{
+    public int Id { get; set; }
+    public int BookId { get; set; }
+    public Book Book { get; set; } = new Book();
```

Comment:
Czemu "= new Book()" ? Robisz konstruktor w którym podajesz książkę której jest to egzemplarz.

Reply:
no nie do końca
```
An unhandled exception occurred while processing the request.
InvalidOperationException: No suitable constructor was found for entity type 'BookCopy'. The following constructors had parameters that could not be bound to properties of the entity type:
Cannot bind 'book' in 'BookCopy(Book book)'
Note that only mapped properties can be bound to constructor parameters. Navigations to related entities, including references to owned types, cannot be bound.
Microsoft.EntityFrameworkCore.Metadata.Internal.ConstructorBindingFactory.GetBindings<T>(T type, Func<IPropertyParameterBindingFactory, T, Type, string, ParameterBinding> bindToProperty, Func<IParameterBindingFactory, T, Type, string, ParameterBinding> bind, out InstantiationBinding constructorBinding, out InstantiationBinding serviceOnlyBinding)
```
to muwi czat
```
To solve this issue, you can modify your BookCopy entity by removing the constructor and instead initializing the properties in the class itself. This way, you don't need a constructor with parameters, and you won't face the issue.

Here's how you can modify your BookCopy entity:

csharp
Download
Copy code
namespace web_library.Book.Entity;

public class BookCopy
{
    public int Id { get; set; }
    public int BookId { get; set; }
    public Book Book { get; set; }
}
In the modified version, we've removed the constructor and added an instance initializer to initialize the properties. Now, when you create a new BookCopy object, you can simply set the properties without the need for a constructor.
```

obgadamy to potem